src/components/Container.tsx

```diff
@@ -136,34 +136,48 @@ export class Container<P extends ContainerProperties, S extends ContainerState>
     get placeholderProps(): PlaceHolderModel {
         return {
             cqPath: this.props.cqPath,
-            placeholderClassNames: Constants.NEW_SECTION_CLASS_NAMES,
+            placeholderClassNames: Constants.NEW_SECTION_CLASS_NAMES
         };
     }
 
-    get placeholderComponent() {
+    get placeholderComponent(): JSX.Element | null {
         if (!this.props.isInEditor) {
             return null;
         }
 
         return <ContainerPlaceholder { ...this.placeholderProps }/>;
     }
 
-    public render() {
+    // eslint-disable-next-line max-lines-per-function
+    public render(): JSX.Element {
         let renderScript;
 
-        if (!this.props.isInEditor && this.props.aemNoDecoration){
+        if (!this.props.isInEditor || this.props.aemNoDecoration) {
```

Comment:
can you please explain why did you change this && to ||

Reply:
given if/else determines if annotation should be included or not

the previous condition `(!this.props.isInEditor && this.props.aemNoDecoration)` means do not show annotation if "not in editor and no decoration flag is present", problem is that `this.props.aemNoDecoration` if not provided (most of the cases) - have no default value, thus it's `undefined` and then we have: `!this.props.isInEditor && undefined` -> `true/false && false` -> `false` thus no matter what's the value of `this.props.isInEditor` you will get annotation if `this.props.aemNoDecoration` is not provided

Reply:
I don't think that this change belongs to that PR.

And I am starting to think that the condition with `&&` is actually correct but incomplete:

# The editor requires the decoration
# If the consumer wants to remove the decoration, they can do it by setting the flag to true
## They will lose the styles and attributes we set via the decoration
# Missing: When doing the same when Server-Side Rendering, then I am expecting `this.props.isInEditor` to be false all the time. And I fear that when the rendition is meant for the Editor and that the customer has set the flag `this.props.aemNoDecoration` to true, we return the wrong rendering.

Reply:
reverted, but this must be revisited as it's buggy at the moment